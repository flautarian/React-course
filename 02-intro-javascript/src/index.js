console.log("Hello world!!!")

const dice = {
    faces: 20,
    getRandomValue: function(){
        // 👇️ get number between min (inclusive) and max (inclusive)
        return Math.floor(Math.random() * (this.faces+ 1));
    }
}

const diceResult =`${dice.getRandomValue()}`;
console.log(diceResult);