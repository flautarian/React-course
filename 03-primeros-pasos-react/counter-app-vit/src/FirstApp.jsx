import React from 'react'
import { Fragment } from 'react';

function getPerson() {
    let name = prompt("cual es tu nombre?") || "leonidas";
    let surname = prompt("cual es tu apellido?") || "deuteronomus";
    let job = prompt("cual es tu oficio?") || "espartano";
    return {
        name: name,
        surname: surname,
        job: job
    };
}

let person = getPerson();

export const FirstApp = () => {
    return (
        <Fragment>
            <p>Nombre:</p>
            <p>-Mi nombre es {person.name}</p>
            <p>-Mi apellido es {person.surname}</p>
            <p>-Y mi trabajo es {person.job}</p>
        </Fragment>
    )
}
