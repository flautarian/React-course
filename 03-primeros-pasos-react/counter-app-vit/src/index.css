html, body {
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1em;
    padding: 70px;
  }
  
  code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
      monospace;
  }
  
button {
  padding: 15px;
  margin: 5px;
  font-size: 1.3em;
}