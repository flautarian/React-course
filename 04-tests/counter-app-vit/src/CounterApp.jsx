import React from 'react'
import { Fragment } from 'react'
import PropTypes from 'prop-types';
import { useState } from 'react';


export const CounterApp = ({value}) => {

    const [ counter, setCounter ] = useState( value );

    const addOne = ( event ) => setCounter((c) => c + 1 );

    const substractOne = ( event ) => setCounter((c) => c - 1 );

    const resetValue = ( event ) => setCounter((c) => c = value);
    
    
    return (
        <Fragment>
            <h1>CounterApp</h1>
            <h2>Valor: { counter } </h2>
            <button onClick={ (event) => addOne( event ) }>+1</button>
            <button onClick={ (event) => substractOne( event ) }>-1</button>
            <button onClick={ (event) => resetValue( event ) }>Reset</button>
        </Fragment>
    )
}

CounterApp.propTypes ={
    value: PropTypes.number.isRequired
}