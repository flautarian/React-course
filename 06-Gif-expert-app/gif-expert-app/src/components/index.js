export * from  './add-category'
export * from  './gif-expert-categories-grid'
export * from  './gif-expert-item'
