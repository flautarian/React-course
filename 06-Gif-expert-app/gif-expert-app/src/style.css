* {
    font-family: Helvetica, Arial, sans-serif;
    background-color: rgb(211, 50, 239);
}

body {
    padding: 60px;
}

input {
    background-color: black;
    border-radius: 5px;
    border: 1px solid rgb(42, 0, 81);
    color: rgb(152, 152, 152);
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
}

input::placeholder {
    color: rgb(87, 87, 87);
}

.center-content {
    margin: 0;
    text-align: center;
}

.glow-on-hover {
    width: 120px;
    height: 25px;
    border: none;
    outline: none;
    color: #fff;
    background: #111;
    cursor: pointer;
    position: relative;
    z-index: 0;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

.glow-on-hover:before {
    content: '';
    background: linear-gradient(45deg, #ff0000, #ff7300, #fffb00, #48ff00, #00ffd5, #002bff, #7a00ff, #ff00c8, #ff0000);
    position: absolute;
    top: -2px;
    left: -2px;
    background-size: 400%;
    z-index: -1;
    filter: blur(5px);
    width: calc(100% + 4px);
    height: calc(100% + 4px);
    animation: glowing 20s linear infinite;
    opacity: 0;
    transition: opacity .3s ease-in-out;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

.glow-on-hover:active {
    color: #000
}

.glow-on-hover:active:after {
    background: transparent;
}

.glow-on-hover:hover:before {
    opacity: 1;
}

.glow-on-hover:after {
    z-index: -1;
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    background: #111;
    left: 0;
    top: 0;
    border-bottom-left-radius: 10px;
    border-bottom-right-radius: 10px;
}

@keyframes glowing {
    0% {
        background-position: 0 0;
    }

    50% {
        background-position: 400% 0;
    }

    100% {
        background-position: 0 0;
    }
}

h1 {
    color: #cccccc;
}

h2 {
    font-size: 1.5rem;
    color: #cccccc;
}

h3 {
    font-size: 3rem;
    margin-bottom: 5px;
}

.card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
    display: flex;
    flex-direction: column;
    height: 0%;
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
}

.card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
}

.card img {
    width: 100%;
}