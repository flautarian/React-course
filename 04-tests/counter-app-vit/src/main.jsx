import React from "react";
import ReactDOM from "react-dom/client";
import { Title } from './Title';
import { FirstApp } from "./FirstApp";
import './index.css';
import { CounterApp } from "./CounterApp";

let valueCounterApp = 10;

ReactDOM.createRoot( document.getElementById('root')).render(
  <React.StrictMode>
    <Title title="Hola!, bienvenido a mi App" subtitle="subtitulo"/>
    <FirstApp/>
    <CounterApp value={valueCounterApp}/>
  </React.StrictMode>
)