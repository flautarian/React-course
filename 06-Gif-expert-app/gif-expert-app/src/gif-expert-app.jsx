import { useState, Fragment } from "react";
import { AddCategory, GifExpertCategoriesGrid } from "./components/index";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'One piece']);

    const onAddCategory = (value) => {
        console.log(value);
        if (!!value && !categories.includes(value) )
            setCategories([value, ...categories]);
    }

    return (
        <Fragment>
            {/* Title */}
            <h1>Gif Expert App</h1>
            {/* Input */}
            <AddCategory onAddCategory={onAddCategory} />
            {/* Results */}
            {categories.map(category => {
                return (
                    <GifExpertCategoriesGrid key={category} category={ category }/>
                );
            })}
            
            
        </Fragment>
    )
}