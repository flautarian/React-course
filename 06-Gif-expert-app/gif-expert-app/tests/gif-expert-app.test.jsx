import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { GifExpertApp } from "../src/gif-expert-app";

describe('gitExpertApp test', () => { 
    test('should match with snapshot', () => {
        const { container } = render( <GifExpertApp/ >);
        expect( container ).toMatchSnapshot();
    });

    test('should AddCategory', async() => {
        render( <GifExpertApp/ >);       
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        fireEvent.input( input, { target: { value: "test" }});
        fireEvent.submit( form );
        await waitFor(
            () => {
                const testResultsLabel = screen.getAllByText('test');
                expect( testResultsLabel ).toBeTruthy();
            },
            {
                timeout: 10000
            }
        );
    });

    test('should not add repeated categories', async() => {
        render( <GifExpertApp/ >);       
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        fireEvent.input( input, { target: { value: "test" }});
        fireEvent.submit( form );
        fireEvent.submit( form );
        await waitFor(
            () => {
                const testResultsLabel = screen.getAllByText('test');
                expect( testResultsLabel ).toBeTruthy();
                expect( testResultsLabel.length ).toBe(1);
            },
            {
                timeout: 10000
            }
        );
    });
 })