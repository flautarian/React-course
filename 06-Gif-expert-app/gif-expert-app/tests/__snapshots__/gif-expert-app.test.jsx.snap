// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gitExpertApp test should match with snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <div
    class="center-content"
  >
    <form
      aria-label="form"
    >
      <div>
        <input
          placeholder="get gifs like...."
          value=""
        />
      </div>
      <button
        class="glow-on-hover"
        type="submit"
      >
        Add Gifs
      </button>
    </form>
  </div>
  <h3>
    One Punch
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
  <h3>
    One piece
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
